inductiva/geometry/cad_object.py

```diff
@@ -0,0 +1,236 @@
+"""Class interface to manipulate CADObjects.
+
+With the goal to manipulate the geometry and spatial properties
+of the objects to be used inside our simulators, this class
+provides a simple interface to read the objects from a native
+CAD file.
+"""
+from typing import List, Literal, Optional, Union
+
+try:
+    import pyvista as pv
+except ImportError:
+    pv = None
+
+from inductiva.utils import optional_deps
+
+
+class CADObject:
+    """Class to manipulate CADObjects."""
+
+    @optional_deps.needs_fluids_extra_deps
+    def __init__(self, filename):
+        """"Initialize the CADObject class."""
+
+        self.filename = filename
+        self.data = pv.read(filename)
+
+    def __repr__(self):
+        """Print information about the CAD object.
+        
+        TODO: Add more information about the object.
+        Existing regions, materials, etc...
+        """
+
+        print("\nCADObject details: ")
+        print(self.data)
+        print(f"\nMesh center: {self.data.center}\n")
+
+        return f"CADObject filename: {self.filename}"
+
+    def translate(self, vector: List[float], inplace: bool = False):
+        """Translate the CADObject by a vector.
+
+        Args:
+            vector: The vector to translate the object.
+            inplace: If True, the object is translated in place.
+        """
+        check_spatial_vector(vector, "Vector")
+
+        print("Translate object by: ", vector)
+        data = self.data.translate(vector, inplace=inplace)
+
+        if inplace:
+            self.data = data
+
+        return self
+
+    def to_origin(self, inplace: bool = False):
+        """Translate the CADObject to the origin.
+
+        Args:
+            inplace: If True, the object is translated in place.
+        """
+
+        print("Translating object to the origin.")
+        data = self.data.translate([-value for value in self.data.center])
+
+        if inplace:
+            self.data = data
+
+        return self
+
+    def to_ground(self, inplace: bool = False):
+        """Translate the CADObject to the ground.
+
+        Args:
+            inplace: If True, the object is translated in place.
+        """
```

Comment:
Add a "Returns" section in the docstring to clarify the return type and behavior:
```
CADObject: The CADObject instance after translating it to the ground.
```